common/src/main/java/com/wynntils/models/gear/GearInfoRegistry.java

```diff
@@ -95,38 +101,51 @@ private void loadRegistry() {
                     .create();
             allMajorIds = majorIdGson.fromJson(majorIdsReader, type);
 
-            // Now we can do the gear DB
-            Download dl = Managers.Net.download(UrlId.DATA_STATIC_GEAR);
-            dl.handleReader(reader -> {
-                Gson gearInfoGson = new GsonBuilder()
-                        .registerTypeHierarchyAdapter(GearInfo.class, new GearInfoDeserializer(allMajorIds))
+            // Now get the obtain info DB
+            Download obtainDl = Managers.Net.download(UrlId.DATA_STATIC_ITEM_OBTAIN);
+            obtainDl.handleReader(obtainReader -> {
+                Type obtainType = new TypeToken<Map<String, List<ItemObtainInfo>>>() {}.getType();
+                Gson obtainGson = new GsonBuilder()
+                        .registerTypeHierarchyAdapter(ItemObtainInfo.class, new ItemObtainInfoDeserializer())
                         .create();
-                WynncraftGearInfoResponse gearInfoResponse =
-                        gearInfoGson.fromJson(reader, WynncraftGearInfoResponse.class);
-
-                // Some entries are test entries etc and should be removed
-                List<GearInfo> registry = gearInfoResponse.items.stream()
-                        .filter(gearInfo -> !INVALID_ENTRIES.contains(gearInfo.name()))
-                        .toList();
-
-                // Create fast lookup maps
-                Map<String, GearInfo> lookupMap = new HashMap<>();
-                Map<String, GearInfo> altLookupMap = new HashMap<>();
-                for (GearInfo gearInfo : registry) {
-                    lookupMap.put(gearInfo.name(), gearInfo);
-                    if (gearInfo.metaInfo().apiName().isPresent()) {
-                        altLookupMap.put(gearInfo.metaInfo().apiName().get(), gearInfo);
-                    }
-                }
+                itemObtainMap = obtainGson.fromJson(obtainReader, obtainType);
 
-                // Make the result visisble to the world
-                gearInfoRegistry = registry;
-                gearInfoLookup = lookupMap;
-                gearInfoLookupApiName = altLookupMap;
+                // Now we can do the gear DB
+                downloadGearRegistry();
```

Comment:
What if the one calling this fails

Reply:
Then we can't build up the gear database properly in any case. :( You could possibly argue that it would be better to have a partially correct gear DB in that case. But then you have to start drawing lines. Is it OK to not have major IDs? How much extra complexity do we accept to have code that would handle an only partially complete gear DB? Etc.